mypy/newsemanal/semanal.py

```diff
@@ -3955,6 +3955,21 @@ def add_symbol(self, name: str, node: SymbolNode, context: Optional[Context],
                                  module_hidden=module_hidden)
         return self.add_symbol_table_node(name, symbol, context)
 
+    def add_symbol_skip_local(self, name: str, node: SymbolNode) -> None:
+        """Same as above, but skipping the local namespace.
+
+        This doesn't check for previous definition and is only used
+        for serialization of method-level classes.
```

Comment:
Should this method be used for other types that can leak, such as regular classes?

Since this whole business is pretty delicate, it would be good to explain this in more detail. Something like this perhaps:

> Classes defined within methods can be exposed through an attribute type, but method-level symbol tables aren't serialized. This method can be used to add such classes to an enclosing, serialized symbol table.

Reply:
Ideally yes, but I decided to separate out the bigger refactoring, see https://github.com/python/mypy/issues/6422

Reply:
I will also add a TODO with a link to the issue.